ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
```

Comment:
Imagine if all the usecases in a large system were built this way, and hundreds of usecases use the same gateway. Suddenly you need to change the name of that gateway. How many files need to change. :boom: :boom: :boom: :boom:  viscosity